### ISSUE TITLE
CT: Right-hand side of 'instanceof' is not an object

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/beers-law-lab#246]
After converting beers-law-lab to ES6 classes, this Error started occurring in CT, and seems to be masking #244:

```
beers-law-lab : phet-io-fuzz : require.js : load
Query: brand=phet-io&phetioStandalone&ea&phetioValidateTandems&fuzz&memoryLimit=1000
Uncaught TypeError: Right-hand side of 'instanceof' is not an object
TypeError: Right-hand side of 'instanceof' is not an object
    at Object.isValidValue (https://bayes.colorado.edu/continuous-testing/snapshot-1582780752950/beers-law-lab/js/concentration/model/SoluteIO.js?bust=1582782417553:65:47)
    at Object.isValueValid (https://bayes.colorado.edu/continuous-testing/snapshot-1582780752950/axon/js/ValidatorDef.js?bust=1582782417553:300:69)
    at validate (https://bayes.colorado.edu/continuous-testing/snapshot-1582780752950/axon/js/validate.js?bust=1582782417553:30:20)
    at new ObjectIO (https://bayes.colorado.edu/continuous-testing/snapshot-1582780752950/tandem/js/types/ObjectIO.js?bust=1582782417553:41:7)
    at new SoluteIO (https://bayes.colorado.edu/continuous-testing/snapshot-1582780752950/beers-law-lab/js/concentration/model/SoluteIO.js?bust=1582782417553:19:3)
    at Solute.initializePhetioObject (https://bayes.colorado.edu/continuous-testing/snapshot-1582780752950/tandem/js/PhetioObject.js?bust=1582782417553:352:30)
    at new PhetioObject (https://bayes.colorado.edu/continuous-testing/snapshot-1582780752950/tandem/js/PhetioObject.js?bust=1582782417553:192:12)
    at new Solute (https://bayes.colorado.edu/continuous-testing/snapshot-1582780752950/beers-law-lab/js/concentration/model/Solute.js?bust=1582782417553:54:7)
    at https://bayes.colorado.edu/continuous-testing/snapshot-1582780752950/beers-law-lab/js/concentration/model/Solute.js?bust=1582782417553:89:22
    at Object.execCb (https://bayes.colorado.edu/continuous-testing/snapshot-1582780752950/sherpa/lib/require-2.3.6.js?bust=1582782417442:1696:33)
id: Bayes Chrome
Approximately 2/26/2020, 10:19:12 PM
```

Here's the line that's failing in SoluteIO. `phet.beersLawLab.Solute` is apparently "not an object". I'm guessing that it's undefined.

```js
SoluteIO.validator = { isValidValue: v => v instanceof phet.beersLawLab.Solute };
```

**Comment by KatieWoe** [comment]
Also in Concentration:
```
concentration : phet-io-fuzz : require.js : load
Query: brand=phet-io&phetioStandalone&ea&phetioValidateTandems&fuzz&memoryLimit=1000
Uncaught TypeError: Right-hand side of 'instanceof' is not an object
TypeError: Right-hand side of 'instanceof' is not an object
    at Object.isValidValue (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/beers-law-lab/js/concentration/model/SoluteIO.js:62:45)
    at Object.isValueValid (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/axon/js/ValidatorDef.js:297:67)
    at validate (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/axon/js/validate.js:27:18)
    at new ObjectIO (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/tandem/js/types/ObjectIO.js:38:5)
    at new SoluteIO (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/beers-law-lab/js/concentration/model/SoluteIO.js:16:1)
    at Solute.initializePhetioObject (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/tandem/js/PhetioObject.js:349:28)
    at new PhetioObject (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/tandem/js/PhetioObject.js:189:10)
    at new Solute (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/beers-law-lab/js/concentration/model/Solute.js:51:5)
    at https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/beers-law-lab/js/concentration/model/Solute.js:86:20
id: Bayes Chrome
Approximately 2/27/2020, 8:13:24 PM
concentration : phet-io-fuzz : require.js : run
Query: brand=phet-io&phetioStandalone&ea&phetioValidateTandems&fuzz&memoryLimit=1000
Uncaught TypeError: Right-hand side of 'instanceof' is not an object
TypeError: Right-hand side of 'instanceof' is not an object
    at Object.isValidValue (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/beers-law-lab/js/concentration/model/SoluteIO.js:62:45)
    at Object.isValueValid (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/axon/js/ValidatorDef.js:297:67)
    at validate (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/axon/js/validate.js:27:18)
    at new ObjectIO (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/tandem/js/types/ObjectIO.js:38:5)
    at new SoluteIO (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/beers-law-lab/js/concentration/model/SoluteIO.js:16:1)
    at Solute.initializePhetioObject (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/tandem/js/PhetioObject.js:349:28)
    at new PhetioObject (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/tandem/js/PhetioObject.js:189:10)
    at new Solute (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/beers-law-lab/js/concentration/model/Solute.js:51:5)
    at https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/beers-law-lab/js/concentration/model/Solute.js:86:20
id: Bayes Chrome
Approximately 2/27/2020, 8:13:24 PM
concentration : phet-io-mirror-inputs-fuzz : require.js : load
Uncaught Error: Uncaught TypeError: Right-hand side of 'instanceof' is not an object
TypeError: Right-hand side of 'instanceof' is not an object
    at Object.isValidValue (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/beers-law-lab/js/concentration/model/SoluteIO.js:62:45)
    at Object.isValueValid (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/axon/js/ValidatorDef.js:297:67)
    at validate (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/axon/js/validate.js:27:18)
    at new ObjectIO (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/tandem/js/types/ObjectIO.js:38:5)
    at new SoluteIO (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/beers-law-lab/js/concentration/model/SoluteIO.js:16:1)
    at Solute.initializePhetioObject (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/tandem/js/PhetioObject.js:349:28)
    at new PhetioObject (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/tandem/js/PhetioObject.js:189:10)
    at new Solute (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/beers-law-lab/js/concentration/model/Solute.js:51:5)
    at https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/beers-law-lab/js/concentration/model/Solute.js:86:20
id: Bayes Chrome
Approximately 2/27/2020, 8:13:24 PM
concentration : phet-io-mirror-inputs-fuzz : require.js : run
Uncaught Error: Uncaught TypeError: Right-hand side of 'instanceof' is not an object
TypeError: Right-hand side of 'instanceof' is not an object
    at Object.isValidValue (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/beers-law-lab/js/concentration/model/SoluteIO.js:62:45)
    at Object.isValueValid (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/axon/js/ValidatorDef.js:297:67)
    at validate (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/axon/js/validate.js:27:18)
    at new ObjectIO (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/tandem/js/types/ObjectIO.js:38:5)
    at new SoluteIO (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/beers-law-lab/js/concentration/model/SoluteIO.js:16:1)
    at Solute.initializePhetioObject (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/tandem/js/PhetioObject.js:349:28)
    at new PhetioObject (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/tandem/js/PhetioObject.js:189:10)
    at new Solute (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/beers-law-lab/js/concentration/model/Solute.js:51:5)
    at https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/beers-law-lab/js/concentration/model/Solute.js:86:20
id: Bayes Chrome
Approximately 2/27/2020, 8:13:24 PM
concentration : phet-io-state-fuzz : require.js : load
Uncaught Error: Uncaught TypeError: Right-hand side of 'instanceof' is not an object
TypeError: Right-hand side of 'instanceof' is not an object
    at Object.isValidValue (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/beers-law-lab/js/concentration/model/SoluteIO.js:62:45)
    at Object.isValueValid (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/axon/js/ValidatorDef.js:297:67)
    at validate (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/axon/js/validate.js:27:18)
    at new ObjectIO (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/tandem/js/types/ObjectIO.js:38:5)
    at new SoluteIO (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/beers-law-lab/js/concentration/model/SoluteIO.js:16:1)
    at Solute.initializePhetioObject (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/tandem/js/PhetioObject.js:349:28)
    at new PhetioObject (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/tandem/js/PhetioObject.js:189:10)
    at new Solute (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/beers-law-lab/js/concentration/model/Solute.js:51:5)
    at https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/beers-law-lab/js/concentration/model/Solute.js:86:20
id: Bayes Chrome
Approximately 2/27/2020, 8:13:24 PM
concentration : phet-io-state-fuzz : require.js : run
Uncaught Error: Uncaught TypeError: Right-hand side of 'instanceof' is not an object
TypeError: Right-hand side of 'instanceof' is not an object
    at Object.isValidValue (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/beers-law-lab/js/concentration/model/SoluteIO.js:62:45)
    at Object.isValueValid (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/axon/js/ValidatorDef.js:297:67)
    at validate (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/axon/js/validate.js:27:18)
    at new ObjectIO (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/tandem/js/types/ObjectIO.js:38:5)
    at new SoluteIO (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/beers-law-lab/js/concentration/model/SoluteIO.js:16:1)
    at Solute.initializePhetioObject (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/tandem/js/PhetioObject.js:349:28)
    at new PhetioObject (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/tandem/js/PhetioObject.js:189:10)
    at new Solute (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/beers-law-lab/js/concentration/model/Solute.js:51:5)
    at https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/beers-law-lab/js/concentration/model/Solute.js:86:20
id: Bayes Chrome
Approximately 2/27/2020, 8:13:24 PM
concentration : phet-io-studio-fuzz : require.js : load
Uncaught Error: Uncaught TypeError: Right-hand side of 'instanceof' is not an object
TypeError: Right-hand side of 'instanceof' is not an object
    at Object.isValidValue (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/beers-law-lab/js/concentration/model/SoluteIO.js:62:45)
    at Object.isValueValid (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/axon/js/ValidatorDef.js:297:67)
    at validate (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/axon/js/validate.js:27:18)
    at new ObjectIO (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/tandem/js/types/ObjectIO.js:38:5)
    at new SoluteIO (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/beers-law-lab/js/concentration/model/SoluteIO.js:16:1)
    at Solute.initializePhetioObject (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/tandem/js/PhetioObject.js:349:28)
    at new PhetioObject (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/tandem/js/PhetioObject.js:189:10)
    at new Solute (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/beers-law-lab/js/concentration/model/Solute.js:51:5)
    at https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/beers-law-lab/js/concentration/model/Solute.js:86:20
id: Bayes Chrome
Approximately 2/27/2020, 8:13:24 PM
concentration : phet-io-studio-fuzz : require.js : run
Uncaught Error: Uncaught TypeError: Right-hand side of 'instanceof' is not an object
TypeError: Right-hand side of 'instanceof' is not an object
    at Object.isValidValue (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/beers-law-lab/js/concentration/model/SoluteIO.js:62:45)
    at Object.isValueValid (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/axon/js/ValidatorDef.js:297:67)
    at validate (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/axon/js/validate.js:27:18)
    at new ObjectIO (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/tandem/js/types/ObjectIO.js:38:5)
    at new SoluteIO (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/beers-law-lab/js/concentration/model/SoluteIO.js:16:1)
    at Solute.initializePhetioObject (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/tandem/js/PhetioObject.js:349:28)
    at new PhetioObject (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/tandem/js/PhetioObject.js:189:10)
    at new Solute (https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/beers-law-lab/js/concentration/model/Solute.js:51:5)
    at https://bayes.colorado.edu/continuous-testing/snapshot-1582859604546/beers-law-lab/js/concentration/model/Solute.js:86:20
id: Bayes Chrome
Approximately 2/27/2020, 8:13:24 PM
```

**Comment by pixelzoom** [comment]
Now that we've converted to ES6 modules, my understanding is that cyclic dependencies are OK (because binding are imported, not references).  So I'm hoping that replacing `phet.beersLawLab.Solute` with `import` will resolve this issue.   @samreid does that sound promising?

**Comment by samreid** [comment]
Yes, bindings are imported, but it is unclear whether it would help in this circumstance.  For instance:

```js
// main.js

import X from './X.js';
import Y from './Y.js';

console.log( new X().getY() );
console.log( new Y().getX() );

// X.js
class X{
 getY(){return new Y();}
}

// Y.js
class Y{
  getX(){return new X();}
}
```

This loads correctly and runs correctly.

Hopefully this advantage will translate to IO types, but I haven't tested it yet.

**Comment by samreid** [comment]
I tested this in Energy Skate Park and it worked great:

```diff
Index: js/energy-skate-park/common/model/ControlPointIO.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- js/energy-skate-park/common/model/ControlPointIO.js	(revision 837f306ed78e7a5405f666ba186332d01a605643)
+++ js/energy-skate-park/common/model/ControlPointIO.js	(date 1582908194159)
@@ -10,6 +10,7 @@
 import validate from '../../../../../axon/js/validate.js';
 import ObjectIO from '../../../../../tandem/js/types/ObjectIO.js';
 import energySkatePark from '../../../energySkatePark.js';
+import ControlPoint from './ControlPoint.js';
 
 class ControlPointIO extends ObjectIO {
 
@@ -42,7 +43,7 @@
 }
 
 ControlPointIO.documentation = 'A control point that can manipulate the track.';
-ControlPointIO.validator = { isValidValue: v => v instanceof phet.energySkatePark.ControlPoint };
+ControlPointIO.validator = { isValidValue: v => v instanceof ControlPoint };
 ControlPointIO.typeName = 'ControlPointIO';
 ObjectIO.validateSubtype( ControlPointIO );
 
```

**Comment by samreid** [comment]
However, this fails:

```js
ControlPointIO.validator = { valueType: ControlPoint };
```

with this error:

> ControlPointIO.js:46 Uncaught ReferenceError: Cannot access 'ControlPoint' before initialization at ControlPointIO.js:46

**Comment by pixelzoom** [comment]
I'm trying this in ph-scale and it's failing.

Solute.js:

```js
19 import SoluteIO from './SoluteIO.js';
62       phetioType: SoluteIO
```

SoluteIO.js:

```js
13 import Solute from './Solute.js';
47 SoluteIO.validator = { isValidValue: value => value instanceof Solute };
```

What am I doing wrong?

**Comment by pixelzoom** [comment]
Maybe it's failing because I have static instances of `Solute`?  `grunt update` error says 

`at http://localhost/~cmalley/GitHub/ph-scale/js/common/model/Solute.js:139:24`

... which is the line number of the first static instance.

**Comment by samreid** [comment]
The problem is not merely that they `Solute` instances are being created statically.  The problem is that they are being created statically *during the definition of `Solute.js`*. By moving them to a separate file, SoluteIO and Solute can reference each other without any problem:

<details>

```diff
Index: js/common/model/Solute.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- js/common/model/Solute.js	(revision e63f80d30cfe54202845060f4a24044a0165b7f4)
+++ js/common/model/Solute.js	(date 1582984502609)
@@ -131,67 +131,5 @@
   }
 }
 
-// 'real world' immutable solutions -------------------------------------------------------
-
-// tandem for all static instances of Solute, which are used across all screens
-const SOLUTES_TANDEM = Tandem.GLOBAL.createTandem( 'model' ).createTandem( 'solutes' );
-
-Solute.DRAIN_CLEANER = new Solute( choiceDrainCleanerString, 13, new Color( 255, 255, 0 ), {
-  colorStopColor: new Color( 255, 255, 204 ),
-  tandem: SOLUTES_TANDEM.createTandem( 'drainCleaner' )
-} );
-
-Solute.HAND_SOAP = new Solute( choiceHandSoapString, 10, new Color( 224, 141, 242 ), {
-  colorStopColor: new Color( 232, 204, 255 ),
-  tandem: SOLUTES_TANDEM.createTandem( 'handSoap' )
-} );
-
-Solute.BLOOD = new Solute( choiceBloodString, 7.4, new Color( 211, 79, 68 ), {
-  colorStopColor: new Color( 255, 207, 204 ),
-  tandem: SOLUTES_TANDEM.createTandem( 'blood' )
-} );
-
-Solute.SPIT = new Solute( choiceSpitString, 7.4, new Color( 202, 240, 239 ), {
-  tandem: SOLUTES_TANDEM.createTandem( 'spit' )
-} );
-
-Solute.WATER = new Solute( Water.name, Water.pH, Water.color, {
-  tandem: SOLUTES_TANDEM.createTandem( 'water' )
-} );
-
-Solute.MILK = new Solute( choiceMilkString, 6.5, new Color( 250, 250, 250 ), {
-  tandem: SOLUTES_TANDEM.createTandem( 'milk' )
-} );
-
-Solute.CHICKEN_SOUP = new Solute( choiceChickenSoupString, 5.8, new Color( 255, 240, 104 ), {
-  colorStopColor: new Color( 255, 250, 204 ),
-  tandem: SOLUTES_TANDEM.createTandem( 'chickenSoup' )
-} );
-
-Solute.COFFEE = new Solute( choiceCoffeeString, 5, new Color( 164, 99, 7 ), {
-  colorStopColor: new Color( 255, 240, 204 ),
-  tandem: SOLUTES_TANDEM.createTandem( 'coffee' )
-} );
-
-Solute.ORANGE_JUICE = new Solute( choiceOrangeJuiceString, 3.5, new Color( 255, 180, 0 ), {
-  colorStopColor: new Color( 255, 242, 204 ),
-  tandem: SOLUTES_TANDEM.createTandem( 'orangeJuice' )
-} );
-
-Solute.SODA = new Solute( choiceSodaString, 2.5, new Color( 204, 255, 102 ), {
-  colorStopColor: new Color( 238, 255, 204 ),
-  tandem: SOLUTES_TANDEM.createTandem( 'soda' )
-} );
-
-Solute.VOMIT = new Solute( choiceVomitString, 2, new Color( 255, 171, 120 ), {
-  colorStopColor: new Color( 255, 224, 204 ),
-  tandem: SOLUTES_TANDEM.createTandem( 'vomit' )
-} );
-
-Solute.BATTERY_ACID = new Solute( choiceBatteryAcidString, 1, new Color( 255, 255, 0 ), {
-  colorStopColor: new Color( 255, 224, 204 ),
-  tandem: SOLUTES_TANDEM.createTandem( 'batteryAcid' )
-} );
-
 phScale.register( 'Solute', Solute );
 export default Solute;
\ No newline at end of file
Index: js/macro/model/MacroModel.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- js/macro/model/MacroModel.js	(revision e63f80d30cfe54202845060f4a24044a0165b7f4)
+++ js/macro/model/MacroModel.js	(date 1582984887786)
@@ -15,7 +15,7 @@
 import Beaker from '../../common/model/Beaker.js';
 import Dropper from '../../common/model/Dropper.js';
 import Faucet from '../../common/model/Faucet.js';
-import Solute from '../../common/model/Solute.js';
+import SelectableSolutes from '../../common/model/SelectableSolutes.js';
 import Solution from '../../common/model/Solution.js';
 import PHScaleConstants from '../../common/PHScaleConstants.js';
 import PHScaleQueryParameters from '../../common/PHScaleQueryParameters.js';
@@ -38,18 +38,18 @@
     // @public solute choices, in order that they'll appear in the combo box
     // The order is alphabetical (English names), see https://github.com/phetsims/ph-scale/issues/101
     this.solutes = [
-      Solute.BATTERY_ACID,
-      Solute.BLOOD,
-      Solute.CHICKEN_SOUP,
-      Solute.COFFEE,
-      Solute.DRAIN_CLEANER,
-      Solute.HAND_SOAP,
-      Solute.MILK,
-      Solute.ORANGE_JUICE,
-      Solute.SODA,
-      Solute.SPIT,
-      Solute.VOMIT,
-      Solute.WATER
+      SelectableSolutes.BATTERY_ACID,
+      SelectableSolutes.BLOOD,
+      SelectableSolutes.CHICKEN_SOUP,
+      SelectableSolutes.COFFEE,
+      SelectableSolutes.DRAIN_CLEANER,
+      SelectableSolutes.HAND_SOAP,
+      SelectableSolutes.MILK,
+      SelectableSolutes.ORANGE_JUICE,
+      SelectableSolutes.SODA,
+      SelectableSolutes.SPIT,
+      SelectableSolutes.VOMIT,
+      SelectableSolutes.WATER
     ];
 
     // @public Beaker, everything else is positioned relative to it. Offset constants were set by visual inspection.
@@ -58,7 +58,7 @@
     // Dropper above the beaker
     const yDropper = this.beaker.position.y - this.beaker.size.height - 15;
     // @public
-    this.dropper = new Dropper( Solute.WATER,
+    this.dropper = new Dropper( SelectableSolutes.WATER,
       new Vector2( this.beaker.position.x - 50, yDropper ),
       new Bounds2( this.beaker.left + 40, yDropper, this.beaker.right - 200, yDropper ), {
         tandem: tandem.createTandem( 'dropper' )
Index: js/common/model/SelectableSolutes.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- js/common/model/SelectableSolutes.js	(date 1582985334103)
+++ js/common/model/SelectableSolutes.js	(date 1582985334103)
@@ -0,0 +1,88 @@
+// Copyright 2019, University of Colorado Boulder
+
+import Color from '../../../../scenery/js/util/Color.js';
+import Tandem from '../../../../tandem/js/Tandem.js';
+import phScaleStrings from '../../ph-scale-strings.js';
+import phScale from '../../phScale.js';
+import Solute from './Solute.js';
+import Water from './Water.js';
+
+const choiceBatteryAcidString = phScaleStrings.choice.batteryAcid;
+const choiceBloodString = phScaleStrings.choice.blood;
+const choiceChickenSoupString = phScaleStrings.choice.chickenSoup;
+const choiceCoffeeString = phScaleStrings.choice.coffee;
+const choiceDrainCleanerString = phScaleStrings.choice.drainCleaner;
+const choiceHandSoapString = phScaleStrings.choice.handSoap;
+const choiceMilkString = phScaleStrings.choice.milk;
+const choiceOrangeJuiceString = phScaleStrings.choice.orangeJuice;
+const choiceSodaString = phScaleStrings.choice.soda;
+const choiceSpitString = phScaleStrings.choice.spit;
+const choiceVomitString = phScaleStrings.choice.vomit;
+
+class SelectableSolutes {
+}
+
+// 'real world' immutable solutions -------------------------------------------------------
+
+// tandem for all static instances of Solute, which are used across all screens
+const SOLUTES_TANDEM = Tandem.GLOBAL.createTandem( 'model' ).createTandem( 'solutes' );
+
+SelectableSolutes.DRAIN_CLEANER = new Solute( choiceDrainCleanerString, 13, new Color( 255, 255, 0 ), {
+  colorStopColor: new Color( 255, 255, 204 ),
+  tandem: SOLUTES_TANDEM.createTandem( 'drainCleaner' )
+} );
+
+SelectableSolutes.HAND_SOAP = new Solute( choiceHandSoapString, 10, new Color( 224, 141, 242 ), {
+  colorStopColor: new Color( 232, 204, 255 ),
+  tandem: SOLUTES_TANDEM.createTandem( 'handSoap' )
+} );
+
+SelectableSolutes.BLOOD = new Solute( choiceBloodString, 7.4, new Color( 211, 79, 68 ), {
+  colorStopColor: new Color( 255, 207, 204 ),
+  tandem: SOLUTES_TANDEM.createTandem( 'blood' )
+} );
+
+SelectableSolutes.SPIT = new Solute( choiceSpitString, 7.4, new Color( 202, 240, 239 ), {
+  tandem: SOLUTES_TANDEM.createTandem( 'spit' )
+} );
+
+SelectableSolutes.WATER = new Solute( Water.name, Water.pH, Water.color, {
+  tandem: SOLUTES_TANDEM.createTandem( 'water' )
+} );
+
+SelectableSolutes.MILK = new Solute( choiceMilkString, 6.5, new Color( 250, 250, 250 ), {
+  tandem: SOLUTES_TANDEM.createTandem( 'milk' )
+} );
+
+SelectableSolutes.CHICKEN_SOUP = new Solute( choiceChickenSoupString, 5.8, new Color( 255, 240, 104 ), {
+  colorStopColor: new Color( 255, 250, 204 ),
+  tandem: SOLUTES_TANDEM.createTandem( 'chickenSoup' )
+} );
+
+SelectableSolutes.COFFEE = new Solute( choiceCoffeeString, 5, new Color( 164, 99, 7 ), {
+  colorStopColor: new Color( 255, 240, 204 ),
+  tandem: SOLUTES_TANDEM.createTandem( 'coffee' )
+} );
+
+SelectableSolutes.ORANGE_JUICE = new Solute( choiceOrangeJuiceString, 3.5, new Color( 255, 180, 0 ), {
+  colorStopColor: new Color( 255, 242, 204 ),
+  tandem: SOLUTES_TANDEM.createTandem( 'orangeJuice' )
+} );
+
+SelectableSolutes.SODA = new Solute( choiceSodaString, 2.5, new Color( 204, 255, 102 ), {
+  colorStopColor: new Color( 238, 255, 204 ),
+  tandem: SOLUTES_TANDEM.createTandem( 'soda' )
+} );
+
+SelectableSolutes.VOMIT = new Solute( choiceVomitString, 2, new Color( 255, 171, 120 ), {
+  colorStopColor: new Color( 255, 224, 204 ),
+  tandem: SOLUTES_TANDEM.createTandem( 'vomit' )
+} );
+
+SelectableSolutes.BATTERY_ACID = new Solute( choiceBatteryAcidString, 1, new Color( 255, 255, 0 ), {
+  colorStopColor: new Color( 255, 224, 204 ),
+  tandem: SOLUTES_TANDEM.createTandem( 'batteryAcid' )
+} );
+
+phScale.register( 'SelectableSolutes', SelectableSolutes );
+export default SelectableSolutes;
\ No newline at end of file
Index: js/common/model/SoluteIO.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- js/common/model/SoluteIO.js	(revision e63f80d30cfe54202845060f4a24044a0165b7f4)
+++ js/common/model/SoluteIO.js	(date 1582985210054)
@@ -10,6 +10,7 @@
 import ObjectIO from '../../../../tandem/js/types/ObjectIO.js';
 import ReferenceIO from '../../../../tandem/js/types/ReferenceIO.js';
 import phScale from '../../phScale.js';
+import Solute from './Solute.js';
 
 // Objects are statically created, use reference equality to look up instances for toStateObject/fromStateObject
 class SoluteIO extends ReferenceIO {
@@ -43,9 +44,7 @@
 
 SoluteIO.documentation = 'a solute';
 SoluteIO.typeName = 'SoluteIO';
-
-//TODO #120 should be instanceof Solute, but require(Solute) is cyclic, and phet.phScale.Solute doesn't exist yet
-SoluteIO.validator = { isValidValue: value => value instanceof Object };
+SoluteIO.validator = { isValidValue: value => value instanceof Solute };
 ObjectIO.validateSubtype( SoluteIO );
 
 phScale.register( 'SoluteIO', SoluteIO );

```

</details>

Note, I also tried to convert SelectableSolutes to `Enumeration.byMap` but got an error that Enumeration values cannot have a "name" attribute.  I would recommend changing that to `soluteName` or equivalent so `Enumeration.byMap` could be used.

**Comment by pixelzoom** [comment]
Fixed in the above commits.  In general, static instances that have an associated IO type need to be factored out into their own .js file.   Otherwise the import (binding) of the IO-type is used to get its value, and that fails due to cyclic dependency.  

I factored out SoluteInstances and BeersLawSolutionInstances, static instances of Solute and BeersLawSolution respectively.   Tested using http://localhost/~cmalley/GitHub/beers-law-lab/beers-law-lab_en.html?brand=phet-io&phetioStandalone&ea&phetioValidateTandems&fuzz&memoryLimit=1000.

Closing.
